src/citrine/jobs/waiting.py

```diff
@@ -0,0 +1,144 @@
+from time import time, sleep
+from pprint import pprint
+from citrine._rest.collection import Collection
+from citrine.informatics.modules import Module
+from citrine.resources.workflow_executions import (
+    WorkflowExecution,
+    WorkflowExecutionStatus,
+)
+
+
+class ConditionTimeoutError(RuntimeError):
+    """Error that is raised when timeout is reached but the checked condition is still False."""
+
+    pass
+
+
+def _print_validation_status(
+    status: str, start_time: float, line_start: str = "", line_end: str = "\r"
+):
+    print(
+        "{}Status = {:<25}Elapsed time".format(line_start, status),
+        " = {str(int(time() - start_time)).rjust(3)}s",
+        end=line_end,
+    )
+
+
+def _pretty_execution_status(status: WorkflowExecutionStatus):
+    status_text = status.status
+    output_text = status_text if status_text != "InProgress" else "In progress"
+    return output_text
+
+
+def _print_execution_status(
+    status: WorkflowExecutionStatus, start_time: float, line_end: str = "\r"
+):
+    print(
+        "Status = {:<25}Elapsed time".format(_pretty_execution_status(status)),
+        " = {}s".format(str(int(time() - start_time)).rjust(3)),
+        end=line_end,
+    )
+
+
+def wait_while_validating(
+    collection: Collection[Module],
+    module: Module,
+    print_status_info: bool = False,
+    timeout: float = 1800.0,
+    interval: float = 3.0,
+) -> Module:
+    """
+    Wait until module is validated.
+
+    Parameters
+    ----------
+    collection : Collection[Module]
+        Collection containing module
+    module : Module
+        Module in Collection
+    print_status_info : bool, optional
+        Whether to print status info, by default False
+    timeout : float, optional
+        Maximum time spent inquiring in seconds, by default 1800.0
+    interval : float, optional
+        Inquiry interval in seconds, by default 3.0
+
+    Returns
+    -------
+    Module
+        Module in Collection after validation
+
+    Raises
+    ------
+    ConditionTimeoutError
+        If fails to validate within timeout
+
+    """
+    def is_validated():
+        status = collection.get(module.uid).status
+        if print_status_info:
+            _print_validation_status(status, start)
```

Comment:
`start` seems to be a forward reference: it isn't actually defined until line 83.  The fact that this works is a testament to the loose nature of python, but its still worth fixing to be less confusing to future developers (and possibly static analysis tools).